docs/contracts/v4/concepts/02-flash-accounting.mdx

```diff
@@ -4,12 +4,97 @@ title: Flash Accounting
+
+This design led to inefficiencies because moving tokens around by calling external functions on their smart contracts - especially in a multi-hop swap - was quite expensive. But also, this was the only design possible since each pool was it's own contract and we needed to move tokens in and out of them to keep the logic sound.
+
+With the singleton architecture, a better design now exists - this is called Flash Accounting(hereinafter FA). It is made even more efficient after the introduction of [Transient Storage](https://eips.ethereum.org/EIPS/eip-1153). FA further reduces the gas cost of trades that cross multiple pools and supports more complex integrations with Uniswap v4.
+
+With FA, tokens are moved into the `PoolManager` and each operation (e.g. swap and liquidity modification) conducted in `PoolManager` updates an internal net balance known as `delta`, **only the final output tokens need to be withdrawn out of the** `PoolManager`.
+
 ## Deltas
 
-(TODO: explain deltas / net deltas)
+### Locking 
+
+To ensure correctness and atomicity in complex operations like a multi-hop swap - v4 uses a locking mechanism. Anytime key actions need to take place within the `PoolManager` - e.g. swaps and liquidity modification - a periphery contract must `unlock` the `PoolManager` first. Then integrators implement the `unlockCallback` and proceed with any of the following actions on the pools:
+
+- swap
+- modifyLiquidity
+- donate
+- take
+- settle
+- mint
+- burn
+
+Note that pool initialization can happen outside the context of unlocking the `PoolManager`.
+
+### Balance Delta
+
+Inside `unlockCallback`, the periphery contract does whatever it needs to do, conduct swaps, modify positions, etc - and finally returns a result. At this point, the `PoolManager` checks to make sure all balances have been settled - and nothing is owed to or from the `PoolManager`. If all looks good, it locks again.
+
+The balances settled above is what we refer as the `delta`, a field held in the unlock state i.e. **debits and credits of assets from the `PoolManager`**. 
+
+Once the `PoolManager` is unlocked, it starts with a blank slate. Then, the user may send the `PoolManager` some tokens for a swap to happen, and this creates a debit to the `PoolManager`. The swap logic is run, which then sends tokens back to the user, creating a credit to the `PoolManager`. In case of a multi-hop swap, multiple more debits and credits can be created here. But at the end, when the `unlockCallback` returns - **the net `delta` must be zero, i.e. no pending debits or credits**, and then only will the transaction succeed. This unlock and call style architecture gives callers maximum flexibility in integrating with the core code.
```

Comment:
```suggestion
```

a bit repetitive and slightly inaccurate.. realistically you should call swap operations first and then send tokens after. doesnt make sense to send tokens first then swsap